src/category_theory/closed/ideal2.lean

```diff
@@ -0,0 +1,412 @@
+import category_theory.limits.shapes.binary_products
```

Comment:
This needs a copyright header.

Reply:
It needs many other things too, hence the tags I picked :)